admin/server.go

```diff
@@ -0,0 +1,55 @@
+package admin
+
+import (
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+
+	"github.com/reddit/baseplate.go/log"
+)
+
+const (
+	DefaultAdminAddr = ":6060"
+)
+
+// ServerArgs contain optional configuration for the admin server.
+type ServerArgs struct {
+	// AdminAddr is a custom address for the admin server. Defaults to port 6060 on localhost if not set.
+	AdminAddr string
+	// HealthCheckFn is the HTTP handler for health check.
+	HealthCheckFn func(w http.ResponseWriter, r *http.Request)
```

Comment:
nit: this is the same as [`http.HandlerFunc`](https://pkg.go.dev/net/http#HandlerFunc) so I'd rather just use that:

```suggestion
	HealthCheckFn http.HandlerFunc
```